### ISSUE TITLE
ykcs11 PIN incorrect

### ISSUE THREAD

**Issue by microshine** [Yubico/yubico-piv-tool#112]
I've got `CKR_PIN_INCORRECT` for `C_LOGIN` function, but I use __correct__ PIN.

If I start 1 instance of app which uses `ykcs11` it works ok. But if I start 2 instance of app I've got PIN error.

![image](https://cloud.githubusercontent.com/assets/7333669/24169419/9aaef0ae-0e8e-11e7-9374-14359c688afb.png)

I use: Yubico NEO
OS: MacOS 10.12
NodeJS: 6.10

Steps:

- run app in terminal (A) 
- enter PIN
![image](https://cloud.githubusercontent.com/assets/7333669/24169912/13e99806-0e90-11e7-925a-cd2b4b946b54.png)
- run app in terminal (A)
- run app in terminal (B)
- enter PIN in terminal (A)
![image](https://cloud.githubusercontent.com/assets/7333669/24170013/6edbd9ae-0e90-11e7-8d84-a6bbe5818bbd.png)

You can try these steps by yourself. You need to install [graphene-pk11](https://www.npmjs.com/package/graphene-pk11) first

```
npm mkdir ykcs11_pin_incorrect
cd ykcs11_pin_incorrect
npm install graphene-pk11 -S
```

add `ykcs11.js` file to `ykcs11_pin_incorrect`

run app
```
node ykcs11
```

ykcs11.js
```javascript
const readline = require('readline');
const graphene = require("graphene-pk11");

const library = "/usr/local/lib/libykcs11.dylib";

const mod = graphene.Module.load(library);

mod.initialize();

// console.log(mod.getSlots()); // Print info about slots
const slot = mod.getSlots(0); // get slot by index 0

const session = slot.open(graphene.SessionFlag.SERIAL_SESSION);

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

rl.question('Login PIN:', (answer) => {
  try {
    session.login(answer);
    console.log("Success");
  } catch (e) {
    console.error(e);
  }

  mod.finalize();

  rl.close();
});
```

**Comment by rmhrisk** [comment]
@klali any help you could offer would be appreciated.

**Comment by a-dma** [comment]
as a rule of thumb, concurrent sessions are not supported. It would help to shed some light if you could enable debugging for the module an re-run your code.

**Comment by qpernil** [comment]
PCSC transaction handling was fixed in ykcs11 from yubico-piv-tool 2.0.0, which was recently released.
Also, I was not able to reproduce this problem using two instances of ssh using ykcs11, so I'm closing this issue now. Please re-open if the problem persists with the current version.
